### PR TITLE
Test Cases to get_item_method in Class ReadProcessor of readlinks plugin

### PR DESCRIPTION
**Pull request by MattSilverio**
Closes #

### Technical

Test cases in test_readlinks.py

### Testing

Following the same steps in README about Running Tests.

```
cd docker/
docker compose exec web make test
```

### CODE REVIEW

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import pytest
+
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:2]:
```suggestion
```

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        if url == 'ebooks/iaid':
+            return {'borrowed': 'true'}
+        return default
+
+
+class MockWebContext:
+    def __init__(self):
+        self.site = MockSiteStore()
+
+
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:18]:
```suggestion
import web
```

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+class TestReadProcessor:
+    @pytest.fixture
+    def read_processor(self):
+        options = {}
+        return readlinks.ReadProcessor.__init__(self, options)
+
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:24]:
```suggestion
```

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    def test_get_item_status_lending_library_true(self):
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:25]:
```suggestion
    def test_get_item_status_lending_library_true(self, mock_site):
```

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        collections = ['inlibrary']
+        subjects = ['In library']
+        readlinks.ReadProcessor.__init__(self, options={'debug_items': True})
+
+        status = readlinks.ReadProcessor.get_item_status(
+            self, 'ekey', 'iaid', collections, subjects
+        )
+        assert status == 'restricted - not inlib'
+
+    def test_get_item_status_in_library_show_true(self):
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:63]:
```suggestion
    def test_get_item_status_in_library_show_true(self, mock_site):
```

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    def test_get_item_status_print_disabled_false(self):
+        collections = ['some other collection']
+        subjects = []
+
+        status = readlinks.ReadProcessor.get_item_status(
+            self, 'ekey', 'iaid', collections, subjects
+        )
+        assert status == 'full access'
+
+    def test_get_item_status_checked_out_true(self, monkeypatch):
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:91]:
```suggestion
    def test_get_item_status_checked_out_true(self, monkeypatch, mock_site):
```

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        collections = ['lendinglibrary']
+        subjects = ['Lending library']
+        readlinks.ReadProcessor.__init__(self, options={})
+
+        monkeypatch.setattr(MockSiteStore, 'get', {'borrowed': 'true'})
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:96]:
```suggestion
        monkeypatch.setattr(
            web.ctx.site.store, 'get', lambda _, __: {'borrowed': 'true'}
        )
```

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+        status = readlinks.ReadProcessor.get_item_status(
+            self, 'ekey', 'iaid', collections, subjects
+        )
+        assert status == 'checked out'
+
+    def test_get_item_status_checked_out_false(self, monkeypatch):
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:103]:
```suggestion
    def test_get_item_status_checked_out_false(self, monkeypatch, mock_site):
```

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        collections = ['lendinglibrary']
+        subjects = ['Lending library']
+        readlinks.ReadProcessor.__init__(self, options={})
+
+        monkeypatch.setattr(MockSiteStore, 'get', {'borrowed': 'false'})
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:108]:
```suggestion
        monkeypatch.setattr(
            web.ctx.site.store, 'get', lambda _, __: {'borrowed': 'false'}
        )
```

**Review by scottbarnes — CHANGES_REQUESTED** [review]
This is really great, @MattSilverio.

Excellent approach with `MockSiteStore` and `MockWebContext`. You were almost there, but it turns out, there is an easy, if hidden, shortcut. :)

There is a `mock_site` pytest `fixture` squirreled away in [openlibrary/mocks/mock_infobase.py](https://github.com/internetarchive/openlibrary/blob/4dc3b66415b291647bedd0ae7f6522ee67ec4f3d/openlibrary/mocks/mock_infobase.py#L367-L400) that makes this a lot easier, as my feedback may show.

That alone gets you most of the way, and then you can `monkeypatch` the `get` method as necessary.

Please let me know if you have any questions.

**Review by MattSilverio — COMMENTED** [review]
Changes applied

File: openlibrary/plugins/books/tests/test_readlinks.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
+from openlibrary.plugins.books import readlinks
+import pytest
```
Comment (scottbarnes) [inline openlibrary/plugins/books/tests/test_readlinks.py:2]:
```suggestion
```

File: openlibrary/plugins/books/tests/test_readlinks.py (hunk 3 above)
Comment (scottbarnes) [reply openlibrary/plugins/books/tests/test_readlinks.py:12]: Was marking this as resolved a mistake? I don't think we need this fixture, but if you disagree, or simply have a question, please let me know. :)

**Comment by scottbarnes** [comment]
This is looking good, @MattSilverio. Please see the comment I un-resolved, and also the one I left.

File: openlibrary/plugins/books/tests/test_readlinks.py (hunk 10 above)
Comment (MattSilverio) [reply openlibrary/plugins/books/tests/test_readlinks.py:2]: Checking the code here, I think the same.

**Review by cclauss — APPROVED** [review]
Looks awesome!  Thanks for doing this.

**Review by scottbarnes — APPROVED** [review]
LGTM! Thanks @MattSilverio and @cclauss

**Pull request merged**